.github/workflows/reusable-build.yml

```diff
@@ -0,0 +1,20 @@
+on:
+  workflow_call:
+
+jobs:
+  reusable-test:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true  # Fetch Hugo themes (true OR recursive)
+          fetch-depth: 0    # Fetch all history for .GitInfo and .Lastmod
+
+      - name: Setup Hugo
+        uses: peaceiris/actions-hugo@v2
+        with:
+          hugo-version: '0.112.3'
+          extended: true
```

Comment:
@discordianfish note that I am re-activating this extended option. Locally it works fine with the same hugo engine version. Is it ok for you? Any specific reason why you commented it out?

Reply:
I was thinking about using this: https://github.com/marketplace/actions/repository-dispatch and calling it from arena and agents after a push in their `main`, what do you think?